components/TARGET_PSA/services/psa_prot_internal_storage/COMPONENT_PSA_SRV_IMPL/pits_impl.cpp

```diff
@@ -49,22 +47,66 @@ const uint8_t base64_coding_table[] = {
     '4', '5', '6', '7', '8', '9', '+', '-'
 };
 
-/*
- * \brief Get default KVStore instance for internal flesh storage
- *
- * \return valid pointer to KVStore
- */
-static KVStore *get_kvstore_instance(void)
+static KVStore *kvstore = NULL;
+
+static void its_init(void)
 {
     KVMap &kv_map = KVMap::get_instance();
-
-    KVStore *kvstore = kv_map.get_internal_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
+    kvstore = kv_map.get_internal_kv_instance(STR_EXPAND(MBED_CONF_STORAGE_DEFAULT_KV));
     if (!kvstore) {
         // Can only happen due to system misconfiguration.
         // Thus considered as unrecoverable error for runtime.
         error("Failed getting kvstore instance\n");
     }
-    return kvstore;
+
+    its_version_t version = { 0, 0 };
+    size_t actual_size = 0;
+    KVStore::info_t kv_info;
+    int status = kvstore->get_info(ITS_VERSION_KEY, &kv_info);
+    if (status != MBED_SUCCESS) {
+        version.major = PSA_ITS_API_VERSION_MAJOR;
+        version.minor = PSA_ITS_API_VERSION_MINOR;
+    } else {
+        if (kv_info.size != sizeof(version)) {
+            error("ITS version data is corrupt");
+        }
+
+        status = kvstore->get(ITS_VERSION_KEY, &version, sizeof(version), &actual_size, 0);
+        if ((status != MBED_SUCCESS)  ||
+                ((status == MBED_SUCCESS) && (actual_size != sizeof(version)))) {
+            error("Could not read ITS version data");
+        }
+    }
+
+    if ((version.major > PSA_ITS_API_VERSION_MAJOR) ||
+            ((version.major == PSA_ITS_API_VERSION_MAJOR) && (version.minor > PSA_ITS_API_VERSION_MINOR))) {
+        error("Downgrading ITS version is not allowed");
+    }
+
+    if ((version.major < PSA_ITS_API_VERSION_MAJOR) ||
+            ((version.major == PSA_ITS_API_VERSION_MAJOR) && (version.minor < PSA_ITS_API_VERSION_MINOR))) {
+        psa_its_status_t migration_status = its_version_migrate(kvstore, &version);
+        if (migration_status != PSA_ITS_SUCCESS) {
+            error("ITS migration failed");
+        }
+
+        version.major = PSA_ITS_API_VERSION_MAJOR;
+        version.minor = PSA_ITS_API_VERSION_MINOR;
+    }
+
+    if (kvstore->set(ITS_VERSION_KEY, &version, sizeof(version), 0) != MBED_SUCCESS) {
+        error("Could not write PSA ITS version");
+    }
+}
+
+void its_deinit(void)
+{
+    kvstore = NULL;
+}
+
+psa_its_status_t its_version_migrate(const KVStore *kv, const its_version_t *version)
```

Comment:
kvstore is not const - as it modifies its internal state when writing to flash.
```suggestion
psa_its_status_t its_version_migrate(KVStore *kv, const its_version_t *version)
```